Firestore/Example/Tests/SpecTests/FSTMockDatastore.mm

```diff
@@ -115,6 +115,11 @@ void WatchQuery(const TargetData& query) override {
     // Snapshot version is ignored on the wire
     TargetData sentTargetData =
         query.WithResumeToken(query.resume_token(), SnapshotVersion::None());
+
```

Comment:
This is where the MockDataStore drops the expectedCount from TargetData as WithResumeToken sets the expectedCount to null.

Reply:
Research: does src have same behaivior: calling WithResumeToken and drop expectedCount?